### PR TITLE
Added 'AllowReflection' to permit or deny access to object.GetType()

### PR DESCRIPTION
**Pull request by AngeloG**
I implemented this for use in my program. Perhaps it will be useful in the main branch as well.

### CODE REVIEW

**Comment by viniciusjarina** [comment]
Can one of the admins verify this patch?

**Comment by AngeloG** [comment]
On second thought, wouldn't a better place for this be in `Metatable.GetMember`?

**Comment by isaacbrodsky** [comment]
It seems to me that a user could still call `Type.GetType(string name)`. Wouldn't this patch still allow that?

**Comment by AngeloG** [comment]
That's true, it would still allow that, but only if the CLR is loaded.

File: Core/NLua/Lua.cs [hunk 1]
```diff
@@ -97,6 +97,12 @@ public class Lua : IDisposable
 		/// </summary>
 		public bool IsExecuting { get { return executing; } }
 
+		private bool _AllowReflection = true;
+		/// <summary>
+		/// Whether to allow or deny access to the GetType() function on .Net objects
+		/// </summary>
+		public bool AllowReflection { get { return _AllowReflection; } set { _AllowReflection = value; } }
```
Comment (viniciusjarina) [inline Core/NLua/Lua.cs]: Why don't use Auto-Implemented Properties?

**Comment by viniciusjarina** [comment]
Also, there are a lot of places where reflection is used inside NLua, I think you only cover the case you hit.

I don't think this is useful at all, NLua relies a lot on reflection. If you are trying to write something more thin and simple, try to use KeraLua only, and manipulate the Lua stack using Lua APIs.

**Pull request closed without merging**